app/src/main/java/com/norvellium/tasky/agenda/presentation/agenda/AgendaViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package com.norvellium.tasky.agenda.presentation.agenda
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.norvellium.tasky.auth.data.local.AuthRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    application: Application,
```

Comment:
I'd avoid passing the app instance to ViewModels, since that way they won't be locally unit testable anymore, instead the tests will need to be instrumented ones and therefore run on an emulator which is much slower (using the normal `ViewModel()` is always enough)

Reply:
Thanks, I made this change in response to your last PR where you suggested the following:

"To get rid of the context, you could return result codes (in form of an enum class for example: ValidationResult.EMAIL_BLANK) and then map these to string resource IDs in your ViewModel."

So I believe I implemented what you suggested (in Registration/Login view models, the agenda stuff was something I whipped together really fast and had copied over). However, It seemed to me that the view model would need a context if it is to resolve string resources. Would the "correct" way be to inject a resource manager into the view model, or how else is it supposed to do this?

When it comes to testing, my resource manager would need to be injected and could utilize a mock context, but I still dont see how to 100% get away from having a context although I do see what you are saying about the test needs to be instrumented and how that should be avoided

Reply:
A resource manager is not the best solution, since it won't reload strings after a config change (which is what you want, since if the user switches the language that should lead to a config change and reload all strings). 
Instead, I use the approach I showed in this video: https://www.youtube.com/watch?v=mB1Lej0aDus